hypothesis-python/tests/cover/test_mock.py

```diff
@@ -21,9 +21,15 @@
 
 from hypothesis import given, strategies as st
 
+from tests.common.utils import skipif_threading
+
 
 @given(thing=st.text())
 @mock.patch("math.atan")
+# mock.patch is thread-unsafe. pytest-run-parallel normally detects this and skips
+# the test, but because we set it inside of @given it can't (or doesn't?) peak
+# inside the AST, resulting in a false negative.
```

Comment:
Can we reorder decorators to fix this?

Or, since it'll affect downstream users too, have `@given()` pass through the attributes etc so that this works?

Reply:
we can't reorder since this test is explicitly testing mocking after `@given`. cf `test_can_mock_outside_given_with_fixture`.

I think the fix is more likely to be improving pytest-run-parallel ast detection than fixing `@given`, but I didn't look into it.